### ISSUE TITLE
sndfile and CDAUDIO.WAD support needs backporting

### ISSUE THREAD

**Issue by Lwmte** [opentomb/OpenTomb#347]
Also, gone with "refactored old master".

**Comment by TeslaRus** [comment]
wad audio file loading had been implemented, only audio scripts should be updated / tuned

**Comment by vvs-** [comment]
Script_GetSoundtrack doesn't allow to omit the file name in every track. And duplicating it a hundred times serve no purpose. So, it might require to modify Lua support code as well.

**Comment by TeslaRus** [comment]
for wad file using script should return only CDAUDIO.WAD and type = WAD, track number are used iside audio module (by track number audio module access to wad header, takes offset to the track and loads track)

**Comment by vvs-** [comment]
And this doesn't work because Script_GetSoundtrack concatenates the track path and file names which are stored in an array. You'd need to put the same file name into each array entry for this to work. And it requires the file names to be present.

**Comment by Lwmte** [comment]
Just not to forget: no any WAD file support exists in current OpenTomb master, nor any support for MS-ADPCM WAV files. There is no any streaming routines except OGG streaming, which currently doesn't work as it should (as it loads WHOLE ogg files into memory on level start).

Solution 1: Brace ourselves and integrate sndfile once again.
Solution 2: Find working code for MS-ADPCM streaming somewhere.

**Comment by vvs-** [comment]
What do you mean? The above Lua script patch actually makes it possible to use WAD file for TR3.

Also, similar patch is needed for Lua scripts to use WAV files for TR4/5. Only file names extensions and method need to be changed. I'm able to successfully use such a patch locally.

It's true that there is no streaming support yet. All files are preloaded at the level start.

**Comment by Lwmte** [comment]
I just mean that preloading soundtrack files on level start instead of streaming couldn't in any way count as "support". Sorry. It's absolutely ridiculous and wrong way to work with long audio files.

**Comment by vvs-** [comment]
@Lwmte Seems that streaming is working now. Should this be closed?

**Comment by TeslaRus** [comment]
This time audio streaming works itself, but it preload ALL audio data to the buffer (it is bad); Only one streaming works correctly: RPL video playing (streaming system); May be it will be more correct to create task for real streaming system (I though about moving all audio streaming format support to rpl code part)

**Comment by vvs-** [comment]
BTW, streams can't be stopped now except when Lara dies. This means that several streams can now be played simultaneously which wasn't intended.

**Comment by TeslaRus** [comment]
Streams stopped by next logic [type filtered]: only one type==background or type==chat may be played, type==one_shot do not stop other one_shot streams

**Comment by vvs-** [comment]
Yes. But `ONCE` streams were stopped before recent audio refactoring. And now it's possible to play several `ONCE` streams at once. This sounds awful and wasn't intended in original Core levels. This isn't obvious in PC version because it just lacks anything besides `LOOP` and `CHAT` streams. But in OT it's now perfectly possible to do that, e.g. in _Caves_.
